### PR TITLE
feat(shorebird_cli): replace `force` flag with `allow-native-diffs` and `allow-asset-diffs`

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Removes the `force` flag on all release and patch commands.

Adds `allow-native-diffs` and `allow-asset-diffs` on patch commands to ignore warnings that native code changes or asset changes were detected between the patch and release builds. 

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/patch/patch_aar_command.dart [hunk 1]
```diff
@@ -52,9 +52,17 @@ of the Android app that is using this module.''',
...
+        help: '''
+Patch even if native code diffs are detected.
+NOTE: this is **not** recommended.''',
+        negatable: false,
+      )
+      ..addFlag(
+        'allow-asset-diffs',
+        help: '''
+Patch even if asset diffs are detected
+NOTE: this is **not** recommended.''',
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/commands/patch/patch_aar_command.dart]: This one is less scary than the native one.  We might want to give context on why or link to docs?

**Review by eseidel — COMMENTED** [review]
I guess my only question is if we should offer any sort of transition period for people, or if we should just bump our minor version number and break folks who are using --force.

File: packages/shorebird_cli/lib/src/commands/patch/patch_aar_command.dart [hunk 2]
```diff
@@ -88,13 +99,17 @@ of the Android app that is using this module.''',
     }
 
     final force = results['force'] == true;
-    final dryRun = results['dry-run'] == true;
-
-    if (force && dryRun) {
-      logger.err('Cannot use both --force and --dry-run.');
+    if (force) {
+      logger
+        ..err(PatchCommand.forceDeprecationErrorMessage)
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/patch/patch_aar_command.dart:104]: Isn't this technically a breaking change and not a deprecation if we're changing the behavior of `--force`? Shouldn't we not exit to maintain backward compatibility here?

File: packages/shorebird_cli/lib/src/commands/patch/patch_command.dart [hunk 3]
```diff
@@ -14,6 +15,33 @@ class PatchCommand extends ShorebirdCommand {
...
+  static const forceDeprecationErrorMessage =
+      'The --force flag has been deprecated';
+
+  static final forceDeprecationExplanation =
+      'If your app has known safe native code or asset changes, you can use '
+      'the ${cyan.wrap('--allow-native-diffs')} or '
+      '${cyan.wrap('--allow-asset-diffs')} '
+      'flags. We do not recommend using these flags unless you are '
+      '${styleBold.wrap('absolutely sure')} that the changes are safe. '
+      'Note: the --force flag is not required for use in CI environments.';
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/patch/patch_command.dart]:
```suggestion
      'Note: the ${cyan.wrap('--force flag')} is not required for use in CI environments.';
```

**Review by felangel — COMMENTED** [review]
LGTM except if we're deprecating the `--force` flag we shouldn't break compatibility in this change and should instead preserve the behavior until the next major release imo.

File: packages/shorebird_cli/lib/src/commands/patch/patch_aar_command.dart (hunk 2 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/patch/patch_aar_command.dart:104]: Discussed offline. Resolving.

File: packages/shorebird_cli/lib/src/commands/patch/patch_aar_command.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/patch/patch_aar_command.dart]: Added more info

**Pull request merged**